transports/http/endpoints/api/merkleroots/endpoints.go

```diff
@@ -25,6 +32,42 @@ func (h *handler) RegisterAPIEndpoints(router *gin.RouterGroup, _ *config.HTTPCo
 	merkle := router.Group("/chain/merkleroot")
 	{
 		merkle.POST("/verify", h.verify)
+		merkle.GET("", h.merkleroots)
+	}
+}
+
+// Merkleroot godoc.
+//
+// @Summary Returns merkleroots for the specified range
+// @Tags merkleroots
+// @Accept */*
+// @Produce json
+// @Success 200 {object} domains.MerkleRootsESKPagedResponse
+// @Router /chain/merkleroot [get]
+// @Param batchSize query string false "Batch size of returned merkleroots"
+// @Param lastEvaluatedKey query string false "Last evaluated merkleroot that client has processed"
+// @Security Bearer
+func (h *handler) merkleroots(c *gin.Context) {
+	batchSize := c.DefaultQuery("batchSize", defaultBatchSize)
+	lastEvaluatedKey := c.Query("lastEvaluatedKey")
+
+	batchSizeInt, err := strconv.Atoi(batchSize)
+	if err != nil || batchSizeInt < 0 {
+		// consider throwing we didn't find value for the given key i would keep it as it is as it tells
+		// the client what we expect so it might be a good thing so they know what kind of data
+		// to send
+		err := helpers.ErrorResponseFromMessage("batchSize must be a positive numeric value", http.StatusBadRequest)
```

Comment:
unneeded comment, 
but yes, the idea is correct, we should inform a user what kind of mistake did he made.

Reply:
removed